activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -335,16 +335,25 @@ def transform_values(*args, &block)
       dup.tap { |hash| hash.transform_values!(*args, &block) }
     end
 
-    def transform_keys(*args, &block)
-      return to_enum(:transform_keys) unless block_given?
-      dup.tap { |hash| hash.transform_keys!(*args, &block) }
+    def transform_keys(hash = nil, &block)
```

Comment:
I'm afraid this changes the behavior a small bit:

```ruby
>> {foo: 1}.transform_keys(nil)
(irb):2:in `transform_keys': no implicit conversion of nil into Hash (TypeError)
```

So we need to assign a custom constant object and check for it to know whetever an argument was passed or not.

Reply:
I'm sorry if I'm not understanding your point correctly, but the behavior of
```
{foo: 1}.with_indifferent_access.transform_keys(nil)
```
seems unchanged to me. It returns an Enumerator on my side both before and after the patch. Maybe you're not calling `HWIA#transform_keys` but Ruby's `Hash#transform_keys`?

Reply:
> Maybe you're not calling `HWIA#transform_keys` but Ruby's `Hash#transform_keys`?

Yes but that's my point. We should behave like `Hash#transform_keys` and not accept `transform_keys(nil)`

Reply:
Ahh, you meant to introduce an incompatibility with old versions of Active Support respecting the compatibility with Ruby. I wasn't thinking about that. But yes, that sounds reasonable because I can hardly imagine any use case of passing in `nil` to this method, and so it's very likely just a bug that should be fixed in their side. I just pushed a separate commit to see if the CI accepts this change, and once it passes, I'll squash it.